phase1/src/helpers/accumulator.rs

```diff
@@ -24,58 +25,135 @@ type AccumulatorElementsRef<'a, E: PairingEngine> = (
     &'a E::G2Affine,
 );
 
-/// Given a public key and the accumulator's digest, it hashes each G1 element
-/// along with the digest, and then hashes it to G2.
-pub(crate) fn compute_g2_s_key<E: PairingEngine>(key: &PublicKey<E>, digest: &[u8]) -> Result<[E::G2Affine; 3]> {
-    Ok([
-        compute_g2_s::<E>(&digest, &key.tau_g1.0, &key.tau_g1.1, 0)?,
-        compute_g2_s::<E>(&digest, &key.alpha_g1.0, &key.alpha_g1.1, 1)?,
-        compute_g2_s::<E>(&digest, &key.beta_g1.0, &key.beta_g1.1, 2)?,
-    ])
-}
-
-/// Reads a list of G1 elements from the buffer to the provided `elements` slice
-/// and then checks that their powers pairs ratio matches the one from the
-/// provided `check` pair
-pub(crate) fn check_power_ratios<E: PairingEngine>(
-    (buffer, compression): (&[u8], UseCompression),
-    (start, end): (usize, usize),
-    elements: &mut [E::G1Affine],
-    check: &(E::G2Affine, E::G2Affine),
-) -> Result<()> {
-    let size = buffer_size::<E::G1Affine>(compression);
-    buffer[start * size..end * size].read_batch_preallocated(&mut elements[0..end - start], compression)?;
-    check_same_ratio::<E>(&power_pairs(&elements[..end - start]), check, "Power pairs")?;
-    Ok(())
-}
+cfg_if! {
```

Comment:
Most of the functions in this file are not WASM compatible. Instead, can you please revert this file, and instead move the `serialize` and `deserialize` functions (along with its corresponding tests) into `phase1/src/helpers/buffers.rs` so they will be compiled in the WASM context?

Separately, to clarify my understanding, can you confirm that these methods are necessary for runtime use and not merely for testing purposes?

Reply:
It is used for testing purposes for now.

Reply:
To make sure we're all on the same page - these functions are not needed in WASM, since we only need contribution in WASM, and are used only in verification. So only in testing in the WASM context, but in runtime in non-WASM context. Is that what you meant @howardwu?

Reply:
Yes precisely, how do you suggest we design testing to support verifying the WASM computation result?

Reply:
Good question. I can only think of awkward ways at the moment :)
Maybe if we share storage between different steps in the test, we could output the WASM contribution and verify with a non-WASM test. Still awkward though.

On the other hand, it'd be a shame to introduce complexity to the code-base to make up for awkward testing.